### PR TITLE
[schema] Allow search individuals by term

### PR DESCRIPTION
**Pull request by mafesan**
As requested on issue https://github.com/chaoss/grimoirelab-sortinghat/issues/355, now `Individuals` query accepts a search term (filter `term`). This term is searched on the fields `name`, `email` and `username` among the different identities from the individuals on the registry.

- [x] Update `Individuals` query filter.
- [x] Unit tests updated.

### CODE REVIEW

**Comment by mafesan** [comment]
I have added some more changes: now this term is searched also among the `name` and `email` fields from the profiles.

**Comment by mafesan** [comment]
I have just rebased the branch with the last version of `muggle` branch.

File: sortinghat/core/schema.py [hunk 1]
```diff
@@ -725,6 +727,25 @@ def resolve_individuals(self, info, filters=None,
...
+
+            # Perform search within profiles
+            query_profiles = Profile.objects.order_by('individual')
+            results = query_profiles.filter(Q(name__icontains=search_term) |
+                                            Q(email__icontains=search_term))
+            result_profiles = [result.individual.mk for result in results]
+
+            # Filter matching individuals by their mk
+            query = query.filter(Q(mk__in=result_uids) |
+                                 Q(mk__in=result_profiles))
```
Comment (sduenas) [inline sortinghat/core/schema.py:748]: Isn't possible to use a [union](https://docs.djangoproject.com/en/3.1/ref/models/querysets/#union)? There should be a way to filter without running several queries at once.

Comment (mafesan) [reply sortinghat/core/schema.py]: @sduenas I managed to apply the filter within the same query without needing a `union`, please check the changes. Thank you!

Comment (mafesan) [reply sortinghat/core/schema.py]: @sduenas Thank you for the pointer to `Subquery` functionality, now I have merged both queries into a single one.

**Review by sduenas — APPROVED** [review]
LGTM

**Pull request merged**